tests/src/main/java/org/hibernate/beanvalidation/tck/tests/methodvalidation/ValidateReturnValueTest.java

```diff
@@ -229,15 +227,13 @@ public void testValidationWithSeveralGroups() throws Exception {
 		);
 
 		assertCorrectConstraintTypes( violations, Size.class, Pattern.class );
-		assertCorrectPathNodeNames(
-				violations,
-				names( methodName, TestUtil.RETURN_VALUE_NODE_NAME ),
-				names( methodName, TestUtil.RETURN_VALUE_NODE_NAME )
-		);
-		assertCorrectPathNodeKinds(
-				violations,
-				kinds( ElementKind.METHOD, ElementKind.RETURN_VALUE ),
-				kinds( ElementKind.METHOD, ElementKind.RETURN_VALUE )
+		assertThat( violations ).containsOnlyPaths(
+				pathWith()
```

Comment:
I have to admit the previous approach was a bit silly as it ignored ordering. Now it makes sure that the type and name etc. of the same path are correct. +1!